projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx

```diff
@@ -152,11 +152,20 @@ class SearchApp extends Component {
 		this.props.initializeQueryValues( {
 			defaultSort: this.props.defaultSort,
 		} );
+		this.showResults();
```

Comment:
Should this only be fired if there's a non-empty search query?

I think I would expect the modal to be closed on my final click of the back button here (note that the last search query "h" also persists in the input):

![2021-02-11 14-43-49 2021-02-11 14_44_20](https://user-images.githubusercontent.com/17325/107594785-b47e0d80-6c77-11eb-820a-46f86be4026a.gif)

Bit of overlap with https://github.com/Automattic/jetpack/pull/18462 here. Making "search query is null" === "modal closed" simplifies things a bit.

Reply:
It might be best to fix this within #18462 instead of doing it separately in this PR. I'll investigate and report back.